Feature/insights alert rules

mytev opened the pull request:
- Add application insights alert rule support
- Fix WebTest `_ResourceId` property

tmiyou committed:
Fix WebTest _ResourceId property

tmiyou committed:
Add application insights alert rule

mytev committed:
Merge branch 'develop' into feature/insights-alert-rules

mbernard commented on src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/Add-ArmApplicationInsightsAlertRuleWebhookAction.ps1:
```diff
@@ -0,0 +1,26 @@
+        $ApplicationInsightsAlertRule,
+        [Parameter(Mandatory)]
+        [string]
+        $ServiceUri,
```
Is this the webhook url? If so, we should rename to Webhook

mytev replied:
Yes it is. Renamed to `Webhook`

mbernard commented on src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1:
```diff
@@ -0,0 +1,79 @@
+        [int]
+        $FailedLocationCount,
+        [string]
+        $WindowSize = "PT5M",
```
There a ValidSet for this

mbernard commented on src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1:
```diff
@@ -0,0 +1,79 @@
+            "Microsoft.Azure.Management.Insights.Models.LocationThresholdRuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.ManagementEventRuleCondition")]
+        [string]
+        $ConditionOdataType,
```
We should rename this to Condition and have a simplified ValidSet and then in the function map to the correct long OdataType

mbernard commented on src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1:
```diff
@@ -0,0 +1,79 @@
+            properties  = @{
+                name        = $Name
+                description = $Description
+                isEnabled   = -not $Disable.ToBool()
```
Disabled*

tmiyou committed:
Address PR comments:
- Add validation to parameters
- Fix typo Disabled
- Simplify validation set for Odata type condition
